cql2pgjson/src/main/java/org/z3950/zing/cql/cql2pgjson/CQL2PgJSON.java

```diff
@@ -472,9 +472,24 @@ private SqlSelect toSql(CQLSortNode node) throws QueryValidationException {
       if (modifiers.cqlSort == CqlSort.DESCENDING) {
         desc = " DESC";
       }  // ASC not needed, it's Postgres' default
-
+      
       IndexTextAndJsonValues vals = getIndexTextAndJsonValues(modifierSet.getBase());
-
+      String sortbyClause  = wrapInLowerUnaccent(vals.indexText);
+      //determine if sort is primary key 
+      if(dbTable != null) {
+        String pkColumnName = dbTable.optString("pkColumnName", /* default = */ "id");
```

Comment:
Can you move both instances of this duplicated code into a method, for example getPkColumnName() that returns the default if dbTable is null?